### PR TITLE
Fix issue when the request doesn't complete

### PR DESCRIPTION
**Pull request by fastlorenzo**

### CODE REVIEW

**Comment by fastlorenzo** [comment]
Fixes #766

File: operator/exporter.py [hunk 1]
```diff
@@ -272,23 +272,23 @@ def collect_all_metrics():
             response = requests.get(
                 'http://'+ pod_details["ip_address"] +':8050/_api/metrics',
                 timeout=1)
+            if response.status_code == 200:
```
Comment (leelavg) [inline operator/exporter.py:275]: - Ideally these statements will not raise an error if `response` is valid
- So, why not assign `None` to `response` before `try` and `return metrics` if `except` block is hit

Just a thought

**Review by vatsa287 — APPROVED** [review]

File: operator/exporter.py [hunk 2]
```diff
@@ -272,23 +272,23 @@ def collect_all_metrics():
...
-            if "nodeplugin" in pod_name:
-                set_nodeplugin_data(response, metrics, pod_name, pod_details)
-
-            if "provisioner" in pod_name:
-                set_provisioner_data(response, metrics, pod_name, pod_details)
-
-            if "server" in pod_name:
-                set_server_data(response, metrics, pod_name, pod_details)
-
+
```
Comment (amarts) [inline operator/exporter.py]:
```suggestion
```

**Comment by amarts** [comment]
DCO failure is because of my edits! The original commit has the signed-off, so we will consider this.

@leelavg any thing else you are looking at?

**Comment by leelavg** [comment]
no, all good!

**Pull request merged**